lib/roadie/inliner.rb

```diff
@@ -119,7 +119,22 @@ def find_head
     def create_style_element(style_blocks, parent)
       return unless parent
       element = Nokogiri::XML::Node.new("style", parent.document)
-      element.content = style_blocks.join("\n")
+
+      # For performance reasons, we should group styles with the same media types within
+      # one media query instead of creating thousands of media queries.
+      # https://github.com/artifex404/media-queries-benchmark
```

Comment:
This sounds well and all, but it also means that specificity will be affected (reordering styles will change the cascades).

Most people would probably not have any issues with this, but some will. I think there should be an option on Document to opt out of this optimization.